### PR TITLE
Add pgid (process group id) as rule condition

### PR DESCRIPTION
**Pull request by wilriker**
I use a slightly modified version of `exec_on_tag.sh` to start applications. This works mostly fine with the exception of a starter application forking into the actual application (e.g. eclipse, IntelliJ Idea, Microsoft Teams). For these cases `pgid` is a lot more usefull as it's the same for all applications. With an adjusted timeout and the `once` modifier removed from the rule this leads to the desired behavior of not only having the splash screen but also the actual application spawn on the tag where it was started.

I recon that getting the `pgid` is mostly not trivial but e.g. my small tool [getpgid](https://github.com/wilriker/getpgid) can be used for this purpose.

### CODE REVIEW

**Comment by t-wissmann** [comment]
Thanks! The diff looks good to me!

There are some minor CI-issues:

1. C headers should be included before C++ headers:
```
+#include <unistd.h>
 #include <climits>
 #include <iostream>
-#include <unistd.h>
```

2. The `numeric_props` array in `tests/test_rules.py` needs to be extended by `pgid`.

3. Can you please also add documentation on it in `doc/herbstluftwm.txt` next to the documentation on `pid`? (Please also mention that `pgid` only works if the assumptions in `pid` are met).

4. While you are at it, you can also extend the `NEWS` file by  a line `new rule condition 'pgid'` :-)

**Comment by wilriker** [comment]
All requested changes applied. :-)

File: doc/herbstluftwm.txt [hunk 1]
```diff
@@ -1174,6 +1174,10 @@ Valid 'properties' are:
     the client's process id (Warning: the pid is not available for every client.
     This only matches if the client sets _NET_WM_PID to the pid itself).
 
++pgid+::
+    this client's process group id. Since the pgid of a window is  is derived from
```
Comment (blarz) [inline doc/herbstluftwm.txt:1178]: typo "is is"

Comment (wilriker) [reply doc/herbstluftwm.txt]: Fixed in 218cbb0.

Comment (blarz) [reply doc/herbstluftwm.txt]: Thanks!

**Comment by wilriker** [comment]
Thanks! :-)

**Pull request merged**